l10n_it_fatturapa_out/wizard/efattura.py

```diff
@@ -261,7 +261,24 @@ def to_xml(self, env):
             # i controlli precedenti dovrebbero escludere errori di sintassi XML
             # with open("/tmp/fatturaout.xml", "wb") as o:
             #    o.write(etree.tostring(root, xml_declaration=True, encoding="utf-8"))
-            raise UserError("\n".join(str(e) for e in errors))
+            # Print error paths, as they can be helpful even for non-technical people
+            error_path_string = "\n- ".join(
+                err.path[err.path.index(":") + 1 :] for err in errors
+            )
+            error = _(
+                "Error processing invoice(s) %(invoices)s.\n\n"
+                "Errors in the following fields:\n- %(error_path_string)s",
+                invoices=", ".join(
+                    inv.display_name for inv in template_values["invoices"]
+                ),
+                error_path_string=error_path_string,
+            )
+            # add details in debug mode
+            if self.env.user.has_group("base.group_no_one"):
```

Comment:
Questo non vuol dire che l'utente è in debug mode, ma che può attivarla.
Per sapere se è attiva puoi usare `user_has_groups` (https://github.com/odoo/odoo/blob/aa2fc9dafb08c0f6e8103a6da3d218fe8233e509/odoo/models.py#L1364) che è in grado di gestire questo caso perché fa:
```py
if group_ext_id == 'base.group_no_one':
    # check: the group_no_one is effective in debug mode only
    if user.has_group(group_ext_id) and request and request.session.debug:
        return False
```

Reply:
Ho imparato qualcosa di nuovo! Grazie

Reply:
Sisi, chiaro. Purtroppo `self` in questo caso non è un recordset (non siamo in una classe che eredita da `models.Model`) quindi dovevo scegliere il recordset più comodo e più "pulito" da andarmi a prendere. Usare l'utente mi sembrava la cosa più elegante e chiara.

(Un'alternativa sarebbe stata usare `template_values["invoices"]` che contiene il recordset delle fatture processate)